pkg/controllers/disruption/drift.go

```diff
@@ -21,33 +21,20 @@ import (
 	"errors"
 	"sort"
 
-	"sigs.k8s.io/controller-runtime/pkg/client"
-
 	"sigs.k8s.io/karpenter/pkg/utils/pretty"
 
 	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	disruptionevents "sigs.k8s.io/karpenter/pkg/controllers/disruption/events"
-	"sigs.k8s.io/karpenter/pkg/controllers/provisioning"
 	"sigs.k8s.io/karpenter/pkg/controllers/provisioning/scheduling"
-	"sigs.k8s.io/karpenter/pkg/controllers/state"
-	"sigs.k8s.io/karpenter/pkg/events"
 )
 
 // Drift is a subreconciler that deletes drifted candidates.
 type Drift struct {
-	kubeClient  client.Client
-	cluster     *state.Cluster
-	provisioner *provisioning.Provisioner
-	recorder    events.Recorder
+	consolidation
 }
 
-func NewDrift(kubeClient client.Client, cluster *state.Cluster, provisioner *provisioning.Provisioner, recorder events.Recorder) *Drift {
-	return &Drift{
-		kubeClient:  kubeClient,
-		cluster:     cluster,
-		provisioner: provisioner,
-		recorder:    recorder,
-	}
+func NewDrift(c consolidation) *Drift {
+	return &Drift{consolidation: c}
```

Comment:
Drift contains consolidation? That doesn't seem right to me. Perhaps we need to change the naming, but I'm also wondering what the need is to encapsulate all of the details of this for drift

Reply:
No need to encapsulate all of the details, we were passing in members of consolidation and this is looking to unify what is required for a new `Method`. If every `Method` takes in consolidation then it could just use what it needs - I think a change to accept a consolidation as a param but then change back to a struct that isn't just `consolidation` is reasonable.